pkg/controller/clone/planner.go

```diff
@@ -305,6 +320,62 @@ func (p *Planner) computeStrategyForSourcePVC(ctx context.Context, args *ChooseS
 	return &strategy, nil
 }
 
+func (p *Planner) computeStrategyForSourceSnapshot(ctx context.Context, args *ChooseStrategyArgs) (*cdiv1.CDICloneStrategy, error) {
+	if ok, err := p.validateTargetStorageClassAssignment(ctx, args); !ok || err != nil {
+		return nil, err
+	}
+
+	// Defaulting to host-assisted clone since it has fewer requirements
+	strategy := cdiv1.CloneStrategyHostAssisted
+
+	// Do volume snapshot validation before checking other specific stuff
+	sourceSnapshot := &snapshotv1.VolumeSnapshot{}
+	exists, err := getResource(ctx, p.Client, args.DataSource.Namespace, args.DataSource.Spec.Source.Name, sourceSnapshot)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		args.Log.V(3).Info("Source Snapshot does not exist, cannot compute strategy")
+		return nil, nil
+	}
+	valid, err := cc.IsSnapshotValidForClone(sourceSnapshot, args.Log)
+	if err != nil || !valid {
+		return nil, err
+	}
+
+	// Do snapshot and storage class validation
+	volumeSnapshotClass := &snapshotv1.VolumeSnapshotClass{}
+	exists, err = getResource(ctx, p.Client, args.DataSource.Namespace, *sourceSnapshot.Spec.VolumeSnapshotClassName, volumeSnapshotClass)
```

Comment:
VolumeSnapshotClasses can come and go (like StorageClasses). It may be better to get the driver from the VolumeSnapshotContent. Then match that driver to the target PVC via StorageClass